Source/CarthageKit/Project.swift

```diff
@@ -483,12 +497,11 @@ public final class Project {
 	/// Checks out the given project into its intended working directory,
 	/// cloning it first if need be.
 	private func checkoutOrCloneProject(project: ProjectIdentifier, atRevision revision: String, submodulesByPath: [String: Submodule]) -> SignalProducer<(), CarthageError> {
-		let repositoryURL = repositoryFileURLForProject(project)
-		let workingDirectoryURL = directoryURL.URLByAppendingPathComponent(project.relativePath, isDirectory: true)
-
-		let checkoutSignal = SignalProducer.attempt { () -> Result<Submodule?, CarthageError> in
+		return cloneOrFetchDependency(project, commitish: revision).flatMap(.Merge) { repositoryURL -> SignalProducer<(), CarthageError> in
+			let workingDirectoryURL = self.directoryURL.URLByAppendingPathComponent(project.relativePath, isDirectory: true)
+			let checkoutSignal = SignalProducer.attempt { () -> Result<Submodule?, CarthageError> in
```

Comment:
Rather than assigning to `checkoutSignal`, this signal producer can just be returned.